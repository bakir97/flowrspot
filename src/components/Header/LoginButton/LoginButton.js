import React from "react";
import styles from "./LoginButton.module.scss";
const LoginButton = ({ openModal }) => {
  const triggerOpenModal = () => {
    openModal("login");
  };
  return (
    <button onClick={triggerOpenModal} className={styles.login}>
      Login
    </button>
  );
};

export default LoginButton;
