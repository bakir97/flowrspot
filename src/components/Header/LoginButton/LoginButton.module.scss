@import "../../../assets/global/styles/colors.scss";
@import "../../../assets/global/styles/mixins.scss";
.login {
  @include font-medium;
  color: $light-orange;
  margin-right: 30px;
  background-color: transparent;
  border: none;
  padding: 0;
  &:hover {
    cursor: pointer;
  }
  &:focus {
    outline: none;
  }
}
