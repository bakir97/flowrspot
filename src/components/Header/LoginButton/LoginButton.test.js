import React from "react";
import { shallow } from "enzyme";
import LoginButton from "./LoginButton";
describe("LoginButton", () => {
  it("should render correctly", () => {
    const component = shallow(<LoginButton />);
    expect(component).toMatchSnapshot();
  });
  it("should call trigger open modal and toggle menu ", () => {
    const mockFn = jest.fn();
    const component = shallow(
      <LoginButton openModal={mockFn} toggleMenu={mockFn} showMenu />
    );
    component.find("button").simulate("click");
    expect(mockFn).toHaveBeenCalledTimes(2);
  });
});
