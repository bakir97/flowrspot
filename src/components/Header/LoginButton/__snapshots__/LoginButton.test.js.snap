// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton should render correctly 1`] = `
<button
  className="login"
  onClick={[Function]}
>
  Login
</button>
`;
